### PR TITLE
Fix misc http issues

### PR DESCRIPTION
**Pull request by kkrishnareddy**
1. remove null key headers from response.
2. no post request if the content length is zero.

### CODE REVIEW

File: common/src/main/java/com/microsoft/identity/common/internal/net/HttpResponse.java [hunk 1]
```diff
@@ -102,4 +102,17 @@ public String toString() {
...
+{        
+    Map<String, List<String>> filteredHeaders = new HashMap<String, List<String>>();        
+    for(String key: headerFields.keySet())        
+    {           
+        if(key!= null)            
+        {                
+            filteredHeaders.put(key,headerFields.get(key));            
+        }
+    }
+    return filteredHeaders;
```
Comment (AdamBJohnsonx) [inline common/src/main/java/com/microsoft/identity/common/internal/net/HttpResponse.java:117]: Why are we doing this in MSAL common?  This seems to be a problem specific to the C++ applications, and should be fixed in the adapter layer there.

**Review by dabbcomputers — CHANGES_REQUESTED** [review]
> 1. remove null key headers from response.
> 2. no post request if the content length is zero.

Don't complete the CR before getting approval from Shane. He wants to have  discussion with Adam first.

File: common/src/main/java/com/microsoft/identity/common/internal/net/HttpResponse.java (hunk 1 above)
Comment (AdamBJohnsonx) [reply common/src/main/java/com/microsoft/identity/common/internal/net/HttpResponse.java:117]: Thing is, this is actually fixed in MSAL-Cpp now.  So I'm not sure we need to do this here.  If we were going to take on moving this to a new field, I'd create one in HttpResponse called mStatusLine and put it there.  The question is - do we need to do this at all?

File: common/src/main/java/com/microsoft/identity/common/internal/net/HttpResponse.java [hunk 2]
```diff
@@ -50,7 +50,7 @@ public HttpResponse(final int statusCode, final String responseBody,
                         final Map<String, List<String>> responseHeaders) {
         mStatusCode = statusCode;
         mResponseBody = responseBody;
-        mResponseHeaders = responseHeaders;
+        mResponseHeaders = filterNullKeyHeaders(responseHeaders);
```
Comment (AdamBJohnsonx) [inline common/src/main/java/com/microsoft/identity/common/internal/net/HttpResponse.java:53]:
```suggestion
        mResponseHeaders = responseHeaders;
```

File: common/src/main/java/com/microsoft/identity/common/internal/net/HttpResponse.java [hunk 3]
```diff
@@ -60,7 +60,7 @@ public HttpResponse(@Nullable final Date date,
         mDate = date;
         mStatusCode = statusCode;
         mResponseBody = responseBody;
-        mResponseHeaders = headerFields;
+        mResponseHeaders = filterNullKeyHeaders(headerFields);
```
Comment (AdamBJohnsonx) [inline common/src/main/java/com/microsoft/identity/common/internal/net/HttpResponse.java:63]:
```suggestion
        mResponseHeaders = headerFields;
```

File: common/src/main/java/com/microsoft/identity/common/internal/net/HttpResponse.java (hunk 1 above)
Comment (AdamBJohnsonx) [inline common/src/main/java/com/microsoft/identity/common/internal/net/HttpResponse.java:117]:
```suggestion
```

**Review by iambmelt — CHANGES_REQUESTED** [review]
I don't think we want this change; as @AdamBJohnsonx states this looks be
1. The wrong repo for this code
2. Already being done in MSAL CPP

As a small aside, the code style isn't consistent with this repo.

Our braces are
```java
void braces() {
    // like this
}
```

and

```java
void braces() 
{
    // Not like this
}
```

And we use `final` wherever possible

**Comment by iambmelt** [comment]
Closing due to inactivity

**Pull request closed without merging**